### PR TITLE
Add v6 preview docs

### LINKED ISSUE #280: Search index missing New-PesterConfiguration codeblock
### Checklist

- [X] Issue has a meaningful title
- [X] I have searched the existing issues. See [issues](https://github.com/pester/docs/issues)

### Link to page

https://pester.dev/docs/commands/New-PesterConfiguration

### What is the issue?

Searching for `SkipRemainingOnFailure` returns no result even though it's documented in the `New-PesterConfiguration` help. I assume it's because codeblocks are excluded from indexing.

### Suggested fix?

Is it possible to include codeblock for that specific page?
Or should we focus on documenting all the options as part of other pages. Ex in this case as a section @ https://pester.dev/docs/usage/skip ?

### LINKED ISSUE #312: Please add v6 (alpha)
### Checklist

- [X] Issue has a meaningful title
- [X] I have searched the existing issues. See [issues](https://github.com/pester/docs/issues)

### Summary of the request

Pester main was rebranded to 6.0.0-alpha1, if possible, please add copy of docs for v6 (alpha), so we can keep it up to date with the changes.

Keep the v5 shown by default please.

### Suggested section

None

### PR DESCRIPTION
**Pull request by fflaten**
Versions v5 docs (`./versioned_docs/v5/*`) and promotes current (`./docs/*`) to v6 preview.

URLs:
- https://pester.dev/docs = v5
- https://pester.dev/docs/v5 = v5 (through Netlify redirect)
- https://pester.dev/docs/v6 = v6 preview docs

Includes `PesterConfiguration` reference in Usage/Configuration page.

Checklist:
- [x] Add Netlify redirect for latest version to stable (/docs/v5/* -> /docs/*) to always support versioned docs URIs
- [x] Add migration docs
- [x] Replace `[PesterConfiguration]::Default` with `New-PesterConfiguration` in docs
- [x] Remove "New in Pester 5.*" comments/tips/badges
- [x] Update/Remove references to "in Pester 5" and similar.
- [x] Generate PesterConfiguration docs in /docs/Configuration file (v6)
  - **Looking for style input**, see https://github.com/pester/docs/issues/280#issuecomment-2166573341
  - Related: pester/pester#2505
- [x] Update `generate-pesterconfiguration-docs.ps1` to support v5 and current (v6)?
  - [x] Actually generate PesterConfiguration docs for v5
- [x] Update `generate-command-reference.ps1` to support v5 and current (v6)?
- [x] Reimplement Github workflow to run scripts above in a clean session
- [ ] Update Assertion-docs (initial draft only)
  - [ ] Introduction to new (Should-*) vs stable (Should -*) assertions?
  - [ ] Update Custom Assertion page to stop recommending registering as should operators?
  - [ ] Rename Should and Assertion reference pages to `... (v5)`?
- [x] Update TestDrive and TestRegistry
- [x] Should we keep /docs/* path for latest stable or always include version like /docs/v5/*, /docs/v6/* to avoid breaking in the future?

Fix #312
Fix pester/pester#2459
Fix #280
Related pester/pester#2464

### CODE REVIEW

**Comment by nohwnd** [comment]
Will it help if I release sooner rather than later?

**Comment by fflaten** [comment]
No hurry. I can build command ref from local build for now.

**Comment by fflaten** [comment]
@bravo-kernel: Thoughts on PR and versioning strategy? Just playing around atm. but considering merging something this week before PSConfEU 🙂

File: .github/workflows/generate-pester-docs.yml
Comment (fflaten) [inline .github/workflows/generate-pester-docs.yml:1]: @bravo-kernel Do you remember why the old manual workflow to draft a update PR was removed (#215) ? With potentially two scripts, I'm thinking it'd be nice to reintroduce it.

**Comment by bravo-kernel** [comment]
@fflaten I have been AFK and will need to find some time to look at this (soon).

File: .github/workflows/generate-pester-docs.yml
Comment (bravo-kernel) [reply .github/workflows/generate-pester-docs.yml:1]: I remember the discussion in ticket https://github.com/pester/Pester/pull/1818 but still think automating version updates would be cool. At least create the PR for it.

Comment (fflaten) [reply .github/workflows/generate-pester-docs.yml:1]: Ahh that makes sense, thanks. Then I'll keep this to quickly draft a PR and simplify it.

**Comment by fflaten** [comment]
> @fflaten awesome job and the preview looks great. Are there any remaining issues you need feedback on?

Thanks 🙌 My initial review thoughts was:
- Versioning / URL scheme: Keep pester.dev/docs for current stable? Always versioned?
  - There's a related question here: Not planning on integrating the release-trigger (yet.. 😉 ), but this will quickly draft a PR to simplify the process.
  - I believe the current iteration with /docs as stable like before with /docs/v<stable> -> /docs redirect might be best. Will allow both types (so module help can refer to versioned URL) while keeping a stable canonical link for search engines so we keep our rank in search results (improved SEO)
- Changes to generate-scripts incl. how to handle sidebar for versioned docs
  - All good? Change/revert something?
  - Was wondering if Docusaurus-module could support versioned docs. E.g. update the versioned sidebar json OR at highlight any added/removed pages between updates. Might be an edge case though. Not sure how many use versioned docs for majors only vs minor versions which a new/removed command would typically require.
- PesterConfiguration docs placement and style 
- Anything else

Should I split this into multiple PRs maybe?

**Comment by bravo-kernel** [comment]
>     * Versioning / URL scheme: Keep pester.dev/docs for current stable? Always versioned?
>       
>       * There's a related question [here](https://github.com/pester/docs/pull/267#issuecomment-2169814740)
>       * I believe the current iteration with /docs as stable like before with /docs/v<stable> -> /docs redirect might be best. Will allow both types (so module help can refer to versioned URL) while keeping a stable canonical link for search engines so we keep our rank in search results (improved SEO)

I am not sure what you are suggesting here but also do not really have a preference. We once chose the current logic with stable being `/docs` and I think it was because it was the Docusaurus default (at that time). Honestly, you have spent a lot of time on this topic so you would probably know best how to proceed here.

>       * Was wondering if Docusaurus-module could support versioned docs. E.g. update the versioned sidebar json OR at highlight any added/removed pages between updates. Might be an edge case though. Not sure how many use versioned docs for majors only vs minor versions which a new/removed command would typically require.

I think we could do the first option. Will need to investigate but perhaps adding a new `-Version` parameter is all that is needed to generate/update the correct versioned sidebar (because we already have https://docusaurus-powershell.vercel.app/docs/commands/New-DocusaurusHelp#-sidebar).

>     * PesterConfiguration docs placement and style
In progress.

> Should I split this into multiple PRs maybe?

Not necessary for me at all, we can always quick fix after merging anyway.

File: .github/workflows/generate-pester-docs.yml
Comment (bravo-kernel) [reply .github/workflows/generate-pester-docs.yml:1]: Another option would be to not depend on the `pester/pester` repo triggering a docs action but... to create a scheduled action in this repo and then just see if there is a new Pester version available and if so, pull that version and update the command reference independently. E.g. check once every night.

**Comment by bravo-kernel** [comment]
New configuration looks great @fflaten

**Comment by fflaten** [comment]
@nohwnd Time for a quick review and maybe merge? Tips to save time:
- Ignore /docs/commands. Nothing special and will be updated on next alpha release.
- Ignore versioned_v5 folder. Mostly two changes affecting it:
  - Updated a mocking helper/snippet, same as v6 page (file: /docs/usage/mocking.mdx)
  - [Added PesterConfiguration reference to Configuration-page](https://deploy-preview-316--pester-docs.netlify.app/docs/usage/configuration#pesterconfiguration-options)
  - Fixed TestDrive-page example to be v5 compatible. Same as v6 page.

File: docs/usage/configuration.mdx [hunk 1]
```diff
@@ -3,30 +3,29 @@ title: Configuration
...
 ShowNavigationMarkers  : Write paths after every block and test, for easy navigation in VSCode. (False, default: False)
+ReturnRawResultObject  : Returns unfiltered result object, this is for development only. Do not rely on this object for additional properties, non-public properties will be renamed without previous notice. (False,
+                         default: False)
 ```
 
-:::tip New in Pester 5.2.0!
-[New-PesterConfiguration](../commands/New-PesterConfiguration) is now available and the recommended way to create the PesterConfiguration-object. It has the added benefit of auto-loading the Pester-module if needed and up-to-date help of available options in the configuration object.
-:::
-
 The configuration object can be constructed either via the Default static property or by casting a hashtable to it. You can also cast a hashtable to any of its sections. Here are three different ways to the same goal:
```
Comment (fflaten) [inline docs/usage/configuration.mdx:22]:
```suggestion
The configuration object can be constructed either using the `New-PesterConfiguration` command or by casting a hashtable to the `PesterConfiguration` type. Here are three different ways to the same goal:
```

**Comment by fflaten** [comment]
Friendly reminder @nohwnd 🙂

**Comment by fflaten** [comment]
@nohwnd Friendly ping 🙂

File: docs/migrations/v5-to-v6.mdx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+description: See this guide for how to update your tests to be compatible with Pester v6.
+---
+
+Pester v6 builds upon the new runtime introduced in v5 and behaves mostly the same which ensures backwards-compatibility. This guide will highlight any necessary steps to update your code and environment to run Pester v6.
+
+## Breaking Changes
+
+### Removed support for PowerShell 3, 4, 5.0, 6, 7.0 and 7.1
+
+To be able to support new functionalit and modernize the codebase, Pester 6.0.0 dropped support for unsupported versions of PowerShell.
```
Comment (nohwnd) [inline docs/migrations/v5-to-v6.mdx:14]:
```suggestion
To be able to support new functionality and modernize the codebase, Pester 6.0.0 dropped support for unsupported versions of PowerShell.
```

**Review by nohwnd — APPROVED** [review]

**Comment by nohwnd** [comment]
Bravo!

**Pull request merged**